Add multi-threading example + complex AOI to timeseries generation example

leothomas opened the pull request:
Screenshots: 

- Complex AOI: France (incl. overseas territories) 
<img width="1124" alt="image" src="https://user-images.githubusercontent.com/11858457/181278301-3f5aa74f-d560-4594-8ffa-7e538befb1e3.png">

- Plot with min/max/mean/stddev: 
<img width="1103" alt="image" src="https://user-images.githubusercontent.com/11858457/181278412-8212a343-97a5-45f9-a485-544f12b5e96c.png">

- Multi-threaded example/explanations:  
<img width="1132" alt="image" src="https://user-images.githubusercontent.com/11858457/181278525-7dd5e314-3005-4b82-b348-f554c8d5bff4.png">

abarciauskas-bgse commented on timeseries-stac-api.ipynb:
```diff
@@ -20,17 +20,16 @@
     "\n",
-    "STAC_API_URL = \"https://staging-stac.delta-backend.com\"\n",
-    "RASTER_API_URL = \"https://staging-raster.delta-backend.com\""
+    "STAC_API_URL = \"https://staging-stac.delta-backend.xyz\"\n",
```
fyi @leothomas I think these urls should be `.com`

abarciauskas-bgse commented:
@leothomas and I discussed updating the URL's to .com, adding titles, adding to the TOC the new notebook(s) and also keeping the existing simple bounding box example

@leothomas let me know when you want me to re-review and thanks agin for all your work on these awesome notebooks!

leothomas commented:
@abarciauskas-bgse this is ready for a re-review. 
- [x] Updated the timeseries notebook to include a simple bounding box example + complex AOI example
- [x] Switched URL's to `staging-*.delta-backend.com`
- [x] Updated the quarto TOC to include the timeseries notebook

abarciauskas-bgse reviewed (approved):
LGTM, I tried to run things locally but am getting an error (unrelated to this) from the Raster API
